### PR TITLE
test: bigquery mock server

### PR DESCRIPTION
**Pull request by PrathameshTugaonkar**
Because

- Essentially because we could mock the BQ server to test it seamlessly

This commit

- tried and researched various approaches and added final approach to mock BQ server

### CODE REVIEW

**Comment by chuang8511** [comment]
Hi @PrathameshTugaonkar 
Thanks for your contribution.
The mock seems good.

Now, could you 
1. Please pre-commit install, you can check the lint before commit.
2. Please follow the git guideline. It is in the 3. point of [sending PR](https://github.com/instill-ai/pipeline-backend/blob/main/.github/CONTRIBUTING.md#sending-prs)
3. Please refer to [json operator test code](https://github.com/instill-ai/pipeline-backend/blob/main/pkg/component/operator/json/v0/component_test.go#L26) to increase the test code coverage. At least, we need to test
  - `func (e *execution) Execute(ctx context.Context, jobs []*base.Job) error {`
  - `func (c *component) GetDefinition(sysVars map[string]any, compConfig *base.ComponentConfig) (*pb.ComponentDefinition, error) {`

Thanks for your contribution again!

**Comment by PrathameshTugaonkar** [comment]
Thanks @chuang8511  for reviewing and providing suggestions.

Is it good to go now? @chuang8511 

Point 3 - Should we take that in another ticket?

**Comment by chuang8511** [comment]
Thanks for your question!

> Point 3 - Should we take that in another ticket?

No, the issue in this ticket mainly lacks the test code for bigquery component.
And, the previous bottleneck is we don't find a good way to mock bigquery.
The end goal here is still adding test for bigquery component.

So, we will still need point 3 in this ticket!

**Comment by PrathameshTugaonkar** [comment]
> Thanks for your question!
> 
> > Point 3 - Should we take that in another ticket?
> 
> No, the issue in this ticket mainly lacks the test code for bigquery component. And, the previous bottleneck is we don't find a good way to mock bigquery. The end goal here is still adding test for bigquery component.
> 
> So, we will still need point 3 in this ticket!

Okay, Definitely will work towards it. Thanks @chuang8511 !

**Comment by PrathameshTugaonkar** [comment]
@chuang8511 Please review. Thanks!

**Comment by chuang8511** [comment]
Hi @PrathameshTugaonkar ,

Thanks for your contribution.

What I mean exactly is - we need to do the test like [here](https://github.com/instill-ai/pipeline-backend/blob/main/pkg/component/operator/json/v0/component_test.go#L179).

In your code, you only test bigquery http fake server. But, you don't interact the bigquery component with the bigquery http fake server.

Recently, I happened to build the similar thing. You can also take a look on [this](https://github.com/instill-ai/pipeline-backend/pull/777/files#diff-b28bb4a40ef5ea4331cf016ba72d7b0d41f80589658e2c1c20b0c0147f4c4766R39) as a reference.

File: pkg/component/data/bigquery/v0/main_test.go [hunk 1]
```diff
@@ -1,3 +1,211 @@
...
+func TestNewClient(t *testing.T) {
+	jsonKey := `{"type": "service_account"}`
+	projectID := "test-project"
+
+	client, err := NewClient(jsonKey, projectID)
+	assert.NoError(t, err)
+	assert.NotNil(t, client)
+}
+
+func TestExecute(t *testing.T) {
```
Comment (chuang8511) [inline pkg/component/data/bigquery/v0/main_test.go:74]: You need to test the interaction between the component and fake bigquery server

**Review by chuang8511 — CHANGES_REQUESTED** [review]
Please check this [comment](https://github.com/instill-ai/pipeline-backend/pull/775#issuecomment-2444904136).

**Pull request closed without merging**